File: 003/3.c
// Find the largest prime factor of a composite number.

#include <stdio.h>

int main () {
	long urPrime = 600851475143;
	long prime = urPrime;
	long teiler = 1;

	while (teiler < urPrime) {
		if (prime % teiler == 0) {
			if (istPrime(teiler)) {
				printf("Primzahlteiler: %ld\n", teiler);
				prime /= teiler;
			}
		}
		teiler++;
	}
	return 0;
}

int istPrime (long zahl) {
	int i = 2;
	while (i < zahl) {

		if (zahl % i == 0) {
			return 0;
		}
		i++;
	}
	return 1;
}

File: 003/3.swift
// The prime factors of 13195 are 5, 7, 13 and 29.
//
// What is the largest prime factor of the number 600851475143?

extension Int {
    func isPrime() -> Bool {
        for i in 2 ..< self {
            if self % i == 0 {
                return false
            }
        }
        return true
    }
}

var num = 600851475143
var factors = [Int]()

while num >= 2 {
    for i in 2 ... num {
        if i.isPrime() && num % i == 0 {
            print("\(num) / \(i)")
            num /= i
            print("= \(num)")
            factors.append(i)
            break
        }
    }
}

print(factors)
